uro98/assignment2/PrintAncestors.java

```diff
@@ -0,0 +1,84 @@
+import java.util.Arrays;
+import java.util.LinkedList;
+
+public class PrintAncestors {
+  
+  /**
+   * Print the ancestors of the node with the given key in a binary tree as a list.
+   * Uses left-to-right depth first search.
+   * An empty list is printed if the node is not found.
+   * @param tree  a generic binary tree
+   * @param key   the target value of a node
+   */
+  public static <T> void printAncestors(BinaryTree<T> tree, T key) {
```

Comment:
This method uses iterative implementation of DFS using queue with non-trivial logic for removal of ancestors based currently visited tree depth. It is absolutely not wrong, but is a bit more complicated than it could.

Optional: what would the code look like with recursive solution? Recursion is a very powerful tool and is used quite a lot in real life, so it is good to be familiar with it.